tests/test_jwt.py

```diff
@@ -187,3 +194,143 @@ def test_roundtrip_explicit_key_id(token_factory):
     certs = {'2': OTHER_CERT_BYTES, '3': PUBLIC_CERT_BYTES}
     payload = jwt.decode(token, certs)
     assert payload['user'] == 'billy bob'
+
+
+class TestCredentials:
+    service_account_email = 'service-account@example.com'
+
+    @pytest.fixture(autouse=True)
+    def credentials(self, signer):
+        self.credentials = jwt.Credentials(
+            signer, self.service_account_email)
+
+    def test_from_service_account_info(self):
+        with open(SERVICE_ACCOUNT_JSON_FILE, 'r') as fh:
+            info = json.load(fh)
+
+        credentials = jwt.Credentials.from_service_account_info(info)
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == info['client_email']
+
+    def test_from_service_account_info_args(self):
+        info = dict(SERVICE_ACCOUNT_INFO)
+
+        credentials = jwt.Credentials.from_service_account_info(
+            info, subject='subject', audience='audience',
+            additional_claims={'meta': 'data'})
+
+        assert credentials._signer.key_id == info['private_key_id']
+        assert credentials._issuer == info['client_email']
+        assert credentials._subject == 'subject'
+        assert credentials._audience == 'audience'
+        assert credentials._additional_claims['meta'] == 'data'
+
+    def test_from_service_account_bad_key(self):
```

Comment:
When I saw `bad_key` I originally assumed you meant bad dictionary key. How about `bad_private_key`?